src/NLog.Extensions.Logging/Config/ExtensionLoggingConfiguration.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using NLog.Config;
+
+namespace NLog.Extensions.Logging
+{
+    /// <summary>
+    /// Configures NLog through Microsoft Extension Configuration section (Ex from appsettings.json)
+    /// </summary>
+    public class ExtensionLoggingConfiguration : LoggingConfigurationParser
+    {
+        private readonly Action<Object> _reloadConfiguration;
+
+        private class LoggingConfigurationElement : ILoggingConfigurationElement
+        {
+            readonly IConfigurationSection _configurationSection;
+            readonly string _nameOverride;
+
+            public bool AutoReload { get; private set; }
+
+            public LoggingConfigurationElement(IConfigurationSection configurationSection, bool topElement, string nameOverride = null)
+            {
+                _configurationSection = configurationSection;
+                _nameOverride = nameOverride;
+                if (topElement && bool.TryParse(configurationSection["autoreload"], out bool autoreload))
+                {
+                    AutoReload = autoreload;
+                }
+            }
+
+            public string Name => _nameOverride ?? _configurationSection.Key;
+
+            public IEnumerable<KeyValuePair<string, string>> Values
+            {
+                get
+                {
+                    var children = _configurationSection.GetChildren();
+                    foreach (var child in children)
+                    {
+                        if (!child.GetChildren().Any())
```

Comment:
Why is this? Because it's not needed if there aren't children?

Reply:
When a child-value contains nested child-values then it is not a simple value, but a child-config-section.